packages/react-form/src/hooks/field/reducer.ts

```diff
@@ -55,62 +55,77 @@ export type FieldAction<Value> =
   | UpdateAction<Value>
   | NewDefaultAction<Value>;
 
-export function reduceField<Value>(
-  state: FieldState<Value>,
-  action: FieldAction<Value>,
-) {
-  switch (action.type) {
-    case 'update': {
-      const newValue = action.payload;
-      const {defaultValue} = state;
-      const isDirty = Array.isArray(defaultValue)
-        ? !shallowArrayComparison(defaultValue, newValue)
-        : defaultValue !== newValue;
-
-      return {
-        ...state,
-        dirty: isDirty,
-        value: newValue,
-        touched: true,
-      };
+export function makeFieldReducer<Value>({
+  dirtyStateComparator,
+}: {
+  dirtyStateComparator: DirtyStateComparator<Value>;
+}): Reducer<FieldState<Value>, FieldAction<Value>> {
+  return (state: FieldState<Value>, action: FieldAction<Value>) => {
+    switch (action.type) {
+      case 'update': {
+        const newValue = action.payload;
+        const {defaultValue} = state;
+        const dirty = dirtyStateComparator(defaultValue, newValue);
```

Comment:
Any reason not to send the `dirtyStateComparator` as part of the action instead of as an option to creating the reducer?

Reply:
Interesting, I didn't consider that. I've thought of it as being something you'd choose for the whole life of the field, which is why I baked it into the reducer. Is that something you think you'd want to choose on a per-action basis?

Reply:
I don't think it is likely you would want to send it on a per-action basis, I guess it just feels slightly more like a piece of information that the `update` action cares about specifically more to me than a fact about the reducer overall, but I could really go either way.

Reply:
I agree that the rest of the reducer doesn't care about it so it is a bit weird to have it available through all the actions. But I'm not really sure how to go about addressing that. Do you have any suggestions?

Reply:
i think what you have here providing it as a factory parameter is fine